query/security/ldap/src/test/groovy/org/codice/ddf/admin/ldap/discover/LdapTestDirectorySettingsSpec.groovy

```diff
@@ -392,4 +392,38 @@ class LdapTestDirectorySettingsSpec extends Specification {
         report.result().getValue()
         ldapConnectionIsClosed
     }
+
+    def 'When useCase = AttributeStore, checkGroupObjectClass and checkGroup should be applied'() {
+        setup:
+        def ldapSettings = initLdapSettings(AUTHENTICATION, true)
+        args = [(LdapConnectionField.DEFAULT_FIELD_NAME)       : noEncryptionLdapConnectionInfo().getValue(),
+                (LdapBindUserInfo.DEFAULT_FIELD_NAME)          : simpleBindInfo().getValue(),
+                (LdapDirectorySettingsField.DEFAULT_FIELD_NAME): ldapSettings.getValue()]
+        action.setValue(args)
+        def utilsMock = Spy(LdapTestConnectionSpec.LdapTestingUtilsMock)
+        action.setTestingUtils(utilsMock)
+
+        when:
+        FunctionReport report = action.getValue()
+
+        then:
+        3 * utilsMock.getLdapQueryResults(*_)
```

Comment:
❗️ This is pretty fragile. Safer would be to assert that the specific query for group object class is not invoked. Something like (this may not be quite correct)...
```groovy
0 * utilsMock.getLdapQueryResults(_, LDAP_SERVER_BASE_GROUP_DN, Filter.equality('objectClass', ldapSettings.groupObjectClass()).toString(), SearchScope.WHOLE_SUBTREE, 1)
```
Use the same logic to assert that the checkGroup query is not executed.

Then in the positive test you simply assert that these queries _are_ run.